main.cpp

```diff
@@ -47,10 +48,11 @@ class CommandLineUI : public GenericUI {
   }
 };
 
-// TODO(sassah): implement a windows UI (WindowUI) that inherits from GenericUI.
-class WindowUI : public GenericUI {
+//.....................................Windows UI Parent Class................................//
+// TODO(sassah): implement a windows UI (WindowsUI) that inherits from GenericUI.
+class WindowsUI : public GenericUI {
```

Comment:
The reason I called it WindowUI is because it is a UI based on a windowed system. Calling it WindowsUI gives the false impression that this will only works for windows, which is not the idea here... No big deal, though.